lib/src/session.dart

```diff
@@ -88,4 +326,272 @@ class Session with ChangeNotifier {
+      }
+      return;
+    }
+
+    // We await for the completer only if the `me` property has been set
+    if ((!_sessionInitialized) && (_me != null)) {
+      if (kDebugMode) {
+        print(
+            '📗 session destroy: !_sessionInitialized, awaiting for _completer.future');
+      }
+      await _completer.future;
+    }
+
+    if (kDebugMode) {
+      print('📗 session destroy: Destroying session');
+    }
+
+    // If the `me` property has not been set, it means that nothing has been done
+    // in the WebView. As a matter of fact we don't even know if the WebView has finished initializing.
+    if (_me != null) {
+      await _execute('session.destroy()');
+    }
+
+    _headlessWebView!.dispose();
+    _headlessWebView = null;
+    _webViewController = null;
+
+    // _completer.isCompleted could be false if we're calling `session.destroy()` beofre setting the `me` property
+    if (!_completer.isCompleted) {
+      _completer.completeError(StateError("The session has been destroyed"));
+    }
+  }
+
+  /// Verifies whether the appId is valid
+  Future<bool> hasValidCredentials() async {
+    if (_headlessWebView == null) {
+      throw StateError(
+          'The hasValidCredentials method cannot be called after destroying the session');
+    }
+
+    if (!_sessionInitialized) {
+      if (_me == null) {
+        throw StateError(
+            'The me property needs to be set for the Session object before calling hasValidCredentials');
+      }
+
+      if (kDebugMode) {
+        print(
+            '📗 session hasValidCredentials: !_sessionInitialized, awaiting for _completer.future');
+      }
+      await _completer.future;
+    }
+
+    if (kDebugMode) {
+      print('📗 session hasValidCredentials: execute');
+    }
+
+    final bool isValid =
+        await _executeAsync('return await session.hasValidCredentials();');
+
+    return isValid;
+  }
+
+  // enablePushNotifications is deliberately omitted, so that we can enable and disable push notifications at will,
+  // without necessarily recreating the ChatBox
+  bool operator ==(Object other) {
+    if (identical(this, other)) {
+      return true;
+    }
+
+    if (!(other is Session)) {
+      return false;
+    }
+
+    if (appId != other.appId) {
+      return false;
+    }
+
+    if (me != other.me) {
+      return false;
+    }
+
+    if (signature != other.signature) {
+      return false;
+    }
+
+    if (onMessage != other.onMessage) {
+      return false;
+    }
+
+    if (unreads != other.unreads) {
+      return false;
+    }
+
+    return true;
+  }
+
+  // me and enablePushNotifications are deliberately omitted, so that this object has the exact same hash regardless of its state
+  int get hashCode => Object.hash(
+        appId,
+        signature,
+        onMessage,
+        unreads,
+      );
+
+  // TODO:
+  // conversation.leave
+  // conversation.sendMessage
+
+  // MAYBE:
+  // onBrowserPermissionDenied
+  // onBrowserPermissionNeeded
```

Comment:
```suggestion
```

We won't need these.

Reply:
thanks for eliminating my doubts

Reply:
I'd also remove from the MAYBE the `conversation.setAttributes` function, because in the Flutter SDK we set everything in the constructor